### PR TITLE
ARROW-8931: [Rust] add lexical sort support to arrow compute kernel

### PR DESCRIPTION
**Pull request by houqp**
In preparation for datafusion sort expression support.

### CODE REVIEW

File: rust/arrow/src/compute/kernels/sort.rs [hunk 1]
```diff
@@ -668,4 +830,300 @@ mod tests {
             ],
         );
     }
+
+    #[test]
+    fn test_lex_sort_single_column() {
+        let input = vec![SortColumn {
+            values: Arc::new(PrimitiveArray::<Int64Type>::from(vec![
+                Some(17),
```
Comment (andygrove) [inline rust/arrow/src/compute/kernels/sort.rs:838]: Could you add `f64::NAN` to the values being tested.

Comment (houqp) [reply rust/arrow/src/compute/kernels/sort.rs:838]: @andygrove this PR currently doesn't support float array due to `Ord` trait constraints. I recommend adding float support as a follow up after we finalize the design in https://github.com/apache/arrow/pull/7193.

Comment (nevi-me) [reply rust/arrow/src/compute/kernels/sort.rs:838]: I got stuck while working on #7193, there'll be merge conflicts between that and this PR, so if this one gets merged first, I'll attempt adding `f64;NAN` when I rebase

**Review by nevi-me — APPROVED** [review]
We can add float lexsort as a separate PR after I complete #7193

**Comment by houqp** [comment]
For those who are interested, here is the datafusion sort expression support commit, which depends on this PR: https://github.com/houqp/arrow/commit/3fbe393c6b6718a70028d311d659e506415666db

**Comment by nevi-me** [comment]
> For those who are interested, here is the datafusion sort expression support commit, which depends on this PR: [houqp@3fbe393](https://github.com/houqp/arrow/commit/3fbe393c6b6718a70028d311d659e506415666db)

This looks great, especially the `concat` kernel. I'll merge this PR tomorrow if there isn't another review, we can always address feedback in subsequent JIRAs.

I also updated the float-sort in #7193, may you please also look at it and see if we'd need to change anything to support floats in lexsort?

**Comment by houqp** [comment]
> I also updated the float-sort in #7193, may you please also look at it and see if we'd need to change anything to support floats in lexsort?

Based on your implementation in #7193, looks like it should be pretty straight forward to add float support to lexsort. We can implement the unwrap to Greater logic in `cmp_value` and relax the Ord constraint for OrdArray. Alternatively, we can change OrdArray into PartialOrdArray that returns `Option<Ordering>`, then add the default to Greater logic into `lexsort` function. Either way will work with minimal code change.

**Pull request closed without merging**